lib/facter/update_initrd_cmd.rb

```diff
@@ -0,0 +1,42 @@
+#
+# == Fact: kmod_update_initrd_cmd
+#
+# Determine which command (if any) must be run to regenerate the initramfs
+# image file for the currently-running kernel.
+require 'facter'
+
+#
+# Function that behaves similarly to the "which" command.
+#
+def which(cmd)
```

Comment:
I would use `Facter::Util::Resolution.which('cmd')`.

Reply:
Yeah, I just found it.  FIGURED there'd be something in Puppet core...